File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/extend_placement_runme.lua
require("import")	-- the import fn
import("extend_placement")	-- import lib into global
ep=extend_placement --alias

-- catching undefined variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

function test_obj(main, suppl)
  assert(main:spam() == 1)
  assert(main:spam("this_is_string") == 2)
  assert(main:spam(5) == 5)
  assert(main:spam(5,6) == 11)
  assert(main:spam(7,8,9) == 15)
  assert(main:spam(suppl,12.0) == 0)
  assert(main:spam(suppl) == 0)
end

foo1 = ep.Foo(0)
foo2 = ep.Foo(1,2)
foo3 = ep.Foo()
test_obj(foo1,foo2)


bar1 = ep.Bar()
bar2 = ep.Bar(5)
test_obj(bar1,bar2)

fti1 = ep.FooTi(0)
fti2 = ep.FooTi(1,2)
fti3 = ep.FooTi()
test_obj(fti1,foo1)

bti1 = ep.BarTi()
bti2 = ep.BarTi(5)
test_obj(bti1,bar1)

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/extend_placement_runme.go
package main

import "./extend_placement"

func main() {
	foo := extend_placement.NewFoo()
	foo = extend_placement.NewFoo(1)
	foo = extend_placement.NewFoo(1, 1)
	foo.Spam()
	foo.Spam("hello")
	foo.Spam(1)
	foo.Spam(1, 1)
	foo.Spam(1, 1, 1)
	foo.Spam(extend_placement.NewFoo())
	foo.Spam(extend_placement.NewFoo(), 1.0)

	bar := extend_placement.NewBar()
	bar = extend_placement.NewBar(1)
	bar.Spam()
	bar.Spam("hello")
	bar.Spam(1)
	bar.Spam(1, 1)
	bar.Spam(1, 1, 1)
	bar.Spam(extend_placement.NewBar())
	bar.Spam(extend_placement.NewBar(), 1.0)

	footi := extend_placement.NewFooTi()
	footi = extend_placement.NewFooTi(1)
	footi = extend_placement.NewFooTi(1, 1)
	footi.Spam()
	footi.Spam("hello")
	footi.Spam(1)
	footi.Spam(1, 1)
	footi.Spam(1, 1, 1)
	footi.Spam(extend_placement.NewFoo())
	footi.Spam(extend_placement.NewFoo(), 1.0)

	barti := extend_placement.NewBarTi()
	barti = extend_placement.NewBarTi(1)
	barti.Spam()
	barti.Spam("hello")
	barti.Spam(1)
	barti.Spam(1, 1)
	barti.Spam(1, 1, 1)
	barti.Spam(extend_placement.NewBar())
	barti.Spam(extend_placement.NewBar(), 1.0)
}
